subprojects/docs/src/docs/userguide/managingTransitiveDependencies.adoc

```diff
@@ -17,6 +17,24 @@
 
 Resolution behavior for transitive dependencies can be customized to a high degree to meet enterprise requirements.
 
+[[sec:dependency_constraints]]
+=== Dependency constraints
+
+Dependency constraints allow you to define the version or the version range of both dependencies declared in the build script and transitive dependencies. It is the preferred method to express constraints that should be applied to all dependencies of a configuration.
+
+++++
+<sample id="dependencyConstraints" dir="userguide/dependencies/dependencyConstraints" title="Define dependency constraints">
+    <sourcefile file="build.gradle" snippet="dependency-constraints"/>
+</sample>
+++++
+
+In the example, all versions are omitted from the dependency declaration. Instead, the versions are defined in the constraints block. The version definition for `commons-codec:1.11` is only taken into account if `commons-codec` is brought in as transitive dependency, since `commons-codec` is not defined as dependency in the project. Otherwise, the constraint has no effect.
+
+Dependency constraints can have different scopes defined by configurations <<sub:scope_of_dependency_configurations,in the same way as dependencies>>.
```

Comment:
I don't quite understand this sentence. We'll need to clarify.

Reply:
I don't want to add a full explanation of configurations here again. So I just want to link the "scope" section in the "declaring dependencies" doc.

I reworded and moved the sentence up to the first paragraph. It's probably not much better...

Do you have a suggestion for better wording?